oaknational/Oak-Web-Application
# PUPIL-362 start replacing buttons with oak components

## Description

An attempt to replace buttons in About us pages with `oak-components` in order to reveal any additional work that needs to be done.

### Remaining work:
- 'Sign up to the newsletter' button on Newsletter form hasn't been replaced as it would also affect the homepage (it uses the same component) and we don't want to have two different styles of buttons on the homepage. It is straightforward to replace but needs  to wait until we replace buttons on the homepage.
- Mobile nav (the one to navigate between About us pages) is still using the old component. This nav is also used on blog and unit listing pages (for categories and filters). The logic slightly differs but most of the logic and design are the same between those three places but currently each of them is using a different component. Ideally this should be ported over to `oak-components` as a base component that can be reused in those three places.
- Desktop nav is using a `OakPrimaryNavItem` component from `oak-components` - this needs to be refined once we have mobile nav ported over. There is `OakPrimaryNav` component in `oak-components` which should be a wrapper to it.
- Leadership and Board pages - 'See bio' button can't be replaced as it looses functionality of returning focus to it once modal is closed (currently when you press 'See bio' on a certain person and then 'X' on modal it will return focus to that person). In addition to this modal needs to be ported over into `oak-components` using relevant buttons from `oak-components` inside the modal 


**After chatting to @kimon-satan  and taking into account above work, we decided not to merge this PR into main which would cause About us pages to have inconsistent buttons and rather treat it as a highlight to what additional button work needs to be done.**

I have added above comments to this document [Replacing SharedComponents in OWA app with oak-components](https://www.notion.so/oaknationalacademy/Replacing-SharedComponents-in-OWA-app-with-oak-components-de39c55b33e947bbbd30888547f5b10e) I've created.


## Issue(s)

Fixes #PUPIL-362

## How to test

1. Go to https://deploy-preview-2222--oak-web-application.netlify.thenational.academy
2. Check all About us pages
3. The buttons should have the styles in line with buttons from `oak-components` and remain functionality of the current site
